### PR TITLE
[CIVP-13354] notebook server extension for uncommitted changes

### PR DESCRIPTION
**Pull request by salilgupta1**

### CODE REVIEW

**Comment by mattyb** [comment]
This looks like it includes an old version of my PR?

File: civis_jupyter_notebooks/extensions/git/uncommitted_changes.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+            response['status'] = 404
+        else:
+            has_changes = False
+            try:
+                has_changes = civis_git.has_uncommitted_changes()
+            except CivisGitError:
+                pass
+
+            response['status'] = 200
+            response['has_uncommitted_changes'] = has_changes
```
Comment (mattyb) [inline civis_jupyter_notebooks/extensions/git/uncommitted_changes.py]: maybe just `dirty`?

File: civis_jupyter_notebooks/extensions/git/uncommitted_changes.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        self.finish(response)
+
+
+def load_jupyter_server_extension(nbapp):
+    nbapp.log.info('Uncommitted Changes Ext. Loaded')
+
+    webapp = nbapp.web_app
+    base_url = webapp.settings['base_url']
+    webapp.add_handlers(".*$", [
+        (ujoin(base_url, r"/git/uncommitted_changes"), UncommittedChangesHandler),
```
Comment (mattyb) [inline civis_jupyter_notebooks/extensions/git/uncommitted_changes.py]: maybe just `dirty`? or even `diff` if you think we might expand the endpoint response in the future

File: civis_jupyter_notebooks/git_utils.py [hunk 3]
```diff
@@ -9,13 +9,31 @@ def __init__(self, repo_url=None, repo_mount_path=None, git_repo_ref=None):
...
 
+    def is_git_enabled(self):
+        if self.repo_url is None:
+            return False
+        return True
+
+    def has_uncommitted_changes(self):
+        try:
+            repo = self.repo()
+            staged_files = len(repo.index.diff("HEAD")) > 0
```
[the remark this answers is not in the document]
Comment (salilgupta1) [reply civis_jupyter_notebooks/git_utils.py]: I could see a case where a user stages files but forgets to commit. I often do that.

[the remark this answers is not in the document]
Comment (mattyb) [reply civis_jupyter_notebooks/git_utils.py]: oh, yeah, that makes sense. I was thinking of the untracked files, but this shouldn't pick those up

File: civis_jupyter_notebooks/extensions/git/uncommitted_changes.py (hunk 2 above)
Comment (salilgupta1) [reply civis_jupyter_notebooks/extensions/git/uncommitted_changes.py]: na, i'm going to keep it as is. We can add a new endpoint/handler if we want to put in new functionality.

File: civis_jupyter_notebooks/extensions/git/uncommitted_changes.py (hunk 1 above)
Comment (salilgupta1) [reply civis_jupyter_notebooks/extensions/git/uncommitted_changes.py]: 👍

**Pull request merged**